api/src/main/java/org/openmrs/module/episodes/dao/impl/EpisodeAttributeTypeDaoImpl.java

```diff
@@ -20,6 +19,10 @@ public class EpisodeAttributeTypeDaoImpl implements EpisodeAttributeTypeDao {
 
     private SessionFactory sessionFactory;
 
+    public void setSessionFactory(SessionFactory sessionFactory) {
```

Comment:
You dont need a setter. You have the constructor taking the sessionFactory. 
If you are going to wire by the moduleApplicationContext.xml - remove the @repository annotation from EpisodeAttributeTypeDaoImpl. Because in the moduleapplicationContext - we have the component scan.